### PR TITLE
Fix dxDateBox rollers ability to select a date in January when the min and max options allow choosing a date in December of the current year and January of the next year  (T584111)

### PR DESCRIPTION
**Pull request by komarovalexander**

### CODE REVIEW

File: js/ui/date_box/ui.date_view.js
Comment (DokaRus) [inline js/ui/date_box/ui.date_view.js]: This PR leads to the changing of the monthRoller options on change every roller. E.g. you change a day from 10 to 11 and monthRoller will be changed too

**Review by DokaRus — APPROVED** [review]
LGTM, wait for one more review before merge

**Review by jsdmitry — APPROVED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/dateView.tests.js
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dateView.tests.js:308]: @komarovalexander 
Why not use a ticket ID in the test subject?

**Pull request merged**
